src/app/components/normal-component/demo-page/demo-card.component.ts

```diff
@@ -1,44 +1,65 @@
-import { AfterViewInit, Component, ComponentFactory, ComponentFactoryResolver, ComponentRef, Input, Type, ViewChild, ViewContainerRef } from '@angular/core';
+import { AfterViewInit, ChangeDetectorRef, Component, ComponentFactoryResolver, Input, ViewChild, ViewContainerRef } from '@angular/core';
 import { AbstractNode } from 'src/app/jscaip/MGPNode';
 import { Utils } from 'src/app/utils/utils';
-import { AbstractGameComponent } from '../../game-components/game-component/GameComponent';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { GameWrapper } from '../../wrapper-components/GameWrapper';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ConnectedUserService } from 'src/app/services/ConnectedUserService';
+import { MessageDisplayer } from 'src/app/services/MessageDisplayer';
+import { Move } from 'src/app/jscaip/Move';
 
 export type DemoNodeInfo = {
     name: string, // The name of the game
-    component: Type<AbstractGameComponent>, // The component
     node: AbstractNode, // The demo node
+    click: MGPOptional<string>, // An element to click
 }
 
 @Component({
     selector: 'app-demo-card',
     template: `<div #board></div>`,
 })
-export class DemoCardComponent implements AfterViewInit {
+export class DemoCardComponent extends GameWrapper<string> implements AfterViewInit {
```

Comment:
DemoCardWrapper ?

Reply:
mouais, la norme d'angular c'est quand même de nommer les composants XXXComponent, donc DemoCardWrapperComponent? Le fait que ça soit un wrapper n'est pas l'élément primordial, c'est un composant qui est une carte de démo, c'est ça qui est important

Reply:
DemoCardWrapperComponent, très bien, cuillère hein ?
Et pour moi Wrapper a sa place dans le nom car c'est fondamentalement son essence, ça wrap un jeu.

Reply:
DemoCardWrapperComponent is geborn!